src/libs/actions/Session.js

```diff
@@ -73,44 +137,60 @@ function signIn(partnerUserID, partnerUserSecret, twoFactorAuthCode = '', exitTo
 
                     setSuccessfulSignInData(createLoginResponse, exitTo);
 
-                    if (credentials && credentials.login) {
-                        // If we have an old login for some reason, we should delete it before storing the new details
+                    // If we have an old login for some reason, we should delete it before storing the new details
+                    if (credentials.login) {
```

Comment:
`credentials.login` got populated after `GetAccountStatus`. I'm guessing that is not the login we want to be deleting here.